src/parser/jobs/sch/modules/Gauge.tsx

```diff
@@ -0,0 +1,105 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import Color from 'color'
+import {JOBS} from 'data/JOBS'
+import {Event} from 'event'
+import {isUndefined} from 'lodash'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge} from 'parser/core/modules/Gauge'
+import {Gauge as CoreGauge} from 'parser/core/modules/Gauge/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const GAUGE_GAIN_AMOUNT = 10
+// Graph color
+const GRAPH_COLOR = Color(JOBS.SCHOLAR.colour)
+// Start at 30 because you can overcap when seraph is out and you can't drain the gauge
+const OVERCAP_SEVERITY = {
+	30: SEVERITY.MINOR,
+	50: SEVERITY.MEDIUM,
+	70: SEVERITY.MAJOR,
+}
+export default class FaerieGauge extends CoreGauge {
+	static override handle = 'fairieGauge'
+	static override title = t('sch.gauge.title')`Faerie Gauge Usage`
+
+	@dependency private suggestions!: Suggestions
+	// Defaults
+	private gauge = this.add(new CounterGauge({
+		chart: {label: 'Faerie Gauge', color: GRAPH_COLOR},
+	}))
+	private fairyOut: boolean = false
+	private actorPets = this.parser.pull.actors.filter(actor => actor.owner != null && actor.owner.id === this.parser.actor.id).map(pet => pet.id)
+	private petHook?: EventHook<{ type: 'action'; action: number; source: string; target: string; timestamp: number; }>
+
+	override initialise() {
+		super.initialise()
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		//Faerie Summon
+		this.petHook = this.addEventHook(playerFilter.type('action').action(oneOf([this.data.actions.SUMMON_EOS.id, this.data.actions.SUMMON_SELENE.id])), this.onSummon)
```

Comment:
You added this to the wrong hook - we always need to check for re-summons (e.g. after death), but the sanity check for marking the pet out if it was out pre-pull (line 44) should only run once / until we see the pet is out.